src/textual/_animator.py

```diff
@@ -241,7 +259,11 @@ def __call__(self) -> None:
             animation_keys = list(self._animations.keys())
             for animation_key in animation_keys:
                 animation = self._animations[animation_key]
-                if animation(animation_time):
+                animation_complete = animation(animation_time)
+                if animation_complete:
+                    completion_callback = animation.on_complete
+                    if completion_callback is not None:
+                        completion_callback()
```

Comment:
I think we will need to use `invoke` here so that you can also pass an awaitable.

Reply:
I updated this to use `invoke`, awaited it, and updated the signature to `async __call__`. The only place that I believe that Animator is called also uses `invoke`, so there was no propagation of `async`.